app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,56 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = Recipe.includes(:user).where(user_id: current_user.id)
+    # @user = User.find_by(id: current_user.id)
+  end
+
+  def show
+    @recipe = Recipe.includes(:user).find(params[:id])
+    # @user = User.find_by(id: current_user.id)
+    @recipe_food = RecipeFood.includes(:food).where(recipe_id: params[:id])
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def public
+    @totals = {}
+    @public_recipes = Recipe.where(public: true).order('created_at DESC')
```

Comment:
- [ ] [OPTIONAL] Following the principle of keeping your controllers thin by putting the necessary logic in the Model, I would suggest that you coud  create a scope in your Recipe model through which you can get all recipes `where(public: true)`. Applying this principle throughout your project code will free up your controllers and make your code easy to read 👍